velox/common/memory/MemoryAllocator.cpp

```diff
@@ -364,11 +371,21 @@ void MemoryAllocator::useHugePages(
 #endif
 }
 
+void MemoryAllocator::setAllocatorFailureMessage(std::string message) {
+  allocatorFailureMessage() = std::move(message);
+}
+
 std::string MemoryAllocator::getAndClearFailureMessage() {
+  auto allocatorMsg = std::move(allocatorFailureMessage());
```

Comment:
```
const std::string allocatorErrMsg = std::move(allocatorFailureMessage());
allocatorFailureMessage().clear();
if (cache() == nullptr) {
   return allocatorErrMsg;
}
return fmt::format("{} {}", allocatorErrMsg, getAndClearCacheFailureMessage());
```

Reply:
With my current implementation I am trying to handle cases where you can have an empty error message either from the allocator or from the cache. Otherwise we might end up with a stray empty space in error messages. For eg, when neither the allocator nor the cache have an error message, OR when only one of them have an error message. `(these can be observed with failed tests in (MemoryPoolTest, MemoryCapExceptions)`

Please let me know if you still prefer I use the suggested implementation for simplicity.